openedx/features/enterprise_support/api.py

```diff
@@ -572,7 +572,12 @@ def consent_needed_for_course(request, user, course_id, enrollment_exists=False)
         )
     else:
         client = ConsentApiClient(user=request.user)
-        current_enterprise_uuid = enterprise_customer_uuid_for_request(request)
+
+        current_enterprise_uuid = None
+        if enterprise_learner_details:
```

Comment:
Do we need this check? We're already in an 'else' clause, having determined that 'enterprise_learner_details' is not None?